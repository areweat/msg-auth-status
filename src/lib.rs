#![warn(
    clippy::unwrap_used,
    missing_docs,
    rust_2018_idioms,
    unused_lifetimes,
    unused_qualifications
)]
#![allow(clippy::single_match, rustdoc::bare_urls)]
#![cfg_attr(all(not(feature = "std"), not(test)), no_std)]
#![doc = include_str!("../README.md")]

#[cfg(all(feature = "alloc", not(feature = "std")))]
extern crate alloc;

//---------------------------------------------------------
// Traits
//---------------------------------------------------------

pub mod traits;

//---------------------------------------------------------
// Error types
//---------------------------------------------------------

pub mod error;

//---------------------------------------------------------
// Authentication-Results & DKIM-Signature etc. pub types
//---------------------------------------------------------

pub mod auth;
pub mod auth_results;
pub mod dkim;
pub mod iprev;
pub mod spf;

//--------------------------------------------------------
// Parsing implementations with type conversions
//--------------------------------------------------------

mod parser;

//--------------------------------------------------------
// Allocating Public convenience API
//--------------------------------------------------------

#[cfg(any(feature = "alloc", feature = "std"))]
pub mod alloc_yes;

//--------------------------------------------------------
// Non-Allocating Public convenience API
//--------------------------------------------------------

#[cfg(feature = "static")]
pub mod alloc_no;
