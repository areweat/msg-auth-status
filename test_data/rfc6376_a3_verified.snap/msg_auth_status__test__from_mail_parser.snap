---
source: src/lib.rs
expression: "&status"
---
MessageAuthStatus {
    auth_results: [],
}
